synapse/storage/data_stores/main/registration.py

```diff
@@ -273,8 +273,7 @@ def delete_account_validity_for_user(self, user_id):
             desc="delete_account_validity_for_user",
         )
 
-    @defer.inlineCallbacks
-    def is_server_admin(self, user):
+    async def is_server_admin(self, user):
```

Comment:
Aren't there more usages of this? You can't `yield` on an async function, though you can `await` on a `Deferred`